### ISSUE TITLE
on 2022-12-14 ,  Kendra claimed that [ Barbastello leucomelas ] should be replace with [ Barbastella leucomelas ] in [ cut:zip:hash://sha256/56caf9620cd58df6fb517dfb7cd01e2e81e54a41eef2f562c5eddfbd70ba6197!/treatments-xml-main/data/C3/05/87/C30587A9A562FF93FF2F350F875ED55F.xml!/b576-597 ]

### ISSUE THREAD

**Issue by jhpoelen** [plazi/community#189]
on 2022-12-14 ,  Kendra claimed that [ Barbastello leucomelas ] should be replace with [ Barbastella leucomelas ] in [ cut:zip:hash://sha256/56caf9620cd58df6fb517dfb7cd01e2e81e54a41eef2f562c5eddfbd70ba6197!/treatments-xml-main/data/C3/05/87/C30587A9A562FF93FF2F350F875ED55F.xml!/b576-597 ]

**Comment by myrmoteras** [comment]
@jhpoelen can you please explain what this long string means and where the error is?

**Comment by myrmoteras** [comment]
fixed

**Comment by myrmoteras** [comment]
@jhpoelen can you please explain what the difference between issues 189-192 is?

**Comment by jhpoelen** [comment]
> @jhpoelen can you please explain what this long string means and where the error is?



```cut:``` prefix indicating the last part (e.g., b576-597) contains a specific byte location or range. Compatible with posix/linux ```cut``` command. 

```zip:``` prefix indicated that a zip entry is addressed within the content. In this case, that entry is: ```treatments-xml-main/data/C3/05/87/C30587A9A562FF93FF2F350F875ED55F.xml```

```hash:``` is followed by the hash type (e.g., sha256) and the hash instance (e.g., 56caf9620cd58df6fb517dfb7cd01e2e81e54a41eef2f562c5eddfbd70ba6197) . The hash, the long hexadecimal sequence, is the unique digital fingerprint of the version of treatments-xml that was retrieved from Github. 


ziphash//sha256/56caf9620cd58df6fb517dfb7cd01e2e81e54a41eef2f562c5eddfbd70ba6197!/treatments-xml-main/data/C3/05/87/C30587A9A562FF93FF2F350F875ED55F.xml!/b576-597

In other words, the long string points to the exact location of the text segment that was the subject of this annotation.

**Comment by jhpoelen** [comment]
In this case, the suspicious name is "Barbastello leucomelas" . 

and, the related treatment is C30587A9A562FF93FF2F350F875ED55F . 

As far as I can tell, all these information elements are currently present in the title and body of the issues.

**Comment by jhpoelen** [comment]
> @jhpoelen can you please explain what the difference between issues 189-192 is?

The difference between the annotations associated with

https://github.com/plazi/community/issues/189
https://github.com/plazi/community/issues/190
https://github.com/plazi/community/issues/191
https://github.com/plazi/community/issues/192

is the location of the subject in the treatment. In the table below, you'll find that the suffix of the reference ids are different. In this case, they point to different text segments of a specific version of a treatment with UUID C30587A9A562FF93FF2F350F875ED55F . 

subjectReferenceId | seeAlso
-- | --
cut:zip:hash://sha256/56caf9620cd58df6fb517dfb7cd01e2e81e54a41eef2f562c5eddfbd70ba6197!/treatments-xml-main/data/C3/05/87/C30587A9A562FF93FF2F350F875ED55F.xml!/b576-597 | https://github.com/plazi/community/issues/189
cut:zip:hash://sha256/56caf9620cd58df6fb517dfb7cd01e2e81e54a41eef2f562c5eddfbd70ba6197!/treatments-xml-main/data/C3/05/87/C30587A9A562FF93FF2F350F875ED55F.xml!/b3054-3075 | https://github.com/plazi/community/issues/190
cut:zip:hash://sha256/56caf9620cd58df6fb517dfb7cd01e2e81e54a41eef2f562c5eddfbd70ba6197!/treatments-xml-main/data/C3/05/87/C30587A9A562FF93FF2F350F875ED55F.xml!/b3550-3571 | https://github.com/plazi/community/issues/191
cut:zip:hash://sha256/56caf9620cd58df6fb517dfb7cd01e2e81e54a41eef2f562c5eddfbd70ba6197!/treatments-xml-main/data/C3/05/87/C30587A9A562FF93FF2F350F875ED55F.xml!/b3714-3735 | https://github.com/plazi/community/issues/192

**Comment by jhpoelen** [comment]
So, it appears that the same subject (e.g., "Barbastello leucomelas") appears four times in the treatment.

**Comment by jhpoelen** [comment]
> fixed

Thanks for addressing the issue. 

Can you please point to the location / version of the treatment that includes the fix?

**Comment by myrmoteras** [comment]
The latest version in the treatment-xml repo, in TB, gbif, etc.

**Comment by myrmoteras** [comment]
> So, it appears that the same subject (e.g., "Barbastello leucomelas") occurs four times in the treatment.

Can you please show me this in the treatment?

**Comment by jhpoelen** [comment]
You can retrieve the text segments using:

```
preston cat 'cut:zip:hash://sha256/56caf9620cd58df6fb517dfb7cd01e2e81e54a41eef2f562c5eddfbd70ba6197!/treatments-xml-main/data/C3/05/87/C30587A9A562FF93FF2F350F875ED55F.xml!/b576-597'
```
yielding 

```
Barbastello leucomelas
```

or 

preston cat 'zip:hash://sha256/56caf9620cd58df6fb517dfb7cd01e2e81e54a41eef2f562c5eddfbd70ba6197!/treatments-xml-main/data/C3/05/87/C30587A9A562FF93FF2F350F875ED55F.xml' | cut -z -b576-597

I can imagine that fancy (web) UI can be hacked on top of this to make it visually a little more intuitive.

Thanks for being patient in resolving the annotations reported by @kephelps, @qgroom and you.

**Comment by myrmoteras** [comment]
Errors are fixed in in TB.

For this, I need to know from you `docId`, `docVersion`, `taxonomicName id`

**Comment by myrmoteras** [comment]
or else, Kendra could ask:  in this treatment https://tb.plazi.org/GgServer/html//C30587A9A562FF93FF2F350F875ED55F the name should be Barbastella  
![image](https://user-images.githubusercontent.com/4609956/211488852-37b068f7-b825-4df3-b6e6-9d99cfdaebbb.png)

**Comment by myrmoteras** [comment]
thanks for the explanation and showing yet another way to look at and interoperate with our data liberated from publications.

**Comment by jhpoelen** [comment]
> For this, I need to know from you docId, docVersion, taxonomicName id

I believe that the requested information can be found in the header of the associated treatments xml file as retrieved via

```
preston cat 'zip:hash://sha256/56caf9620cd58df6fb517dfb7cd01e2e81e54a41eef2f562c5eddfbd70ba6197!/treatments-xml-main/data/C3/05/87/C30587A9A562FF93FF2F350F875ED55F.xml' | head -n1
```


```
<document ID-DOI="http://doi.org/10.5281/zenodo.7353060" ID-ISBN="1-56098-217-9" ID-Zenodo-Dep="7353060" approvalRequired="4" approvalRequired_for_document="2" approvalRequired_for_matCits="1" approvalRequired_for_originalDoi="1" checkinTime="1667534087130" checkinUser="GgServerImporter" docAuthor="Karl F. Koopman" docDate="1993" docId="C30587A9A562FF93FF2F350F875ED55F" docLanguage="en" docName="MammalSpeciesofTheWorld.1993.Chiroptera.137-241.pdf.imd" docOrigin="Mammal Species of the World (2 nd Edition), Washington and London: Smithsonian Institution Press" docTitle="Barbastello leucomelas" docType="treatment" docVersion="3" lastPageNumber="199" masterDocId="3F3CFFD1A55CFFADFFDC34768324D72D" masterDocTitle="Order Chiroptera" masterLastPageNumber="241" masterPageNumber="137" pageNumber="199" updateTime="1669256093699" updateUser="ExternalLinkService">
```

Please confirm that this is the information you need.

**Comment by jhpoelen** [comment]
PS not quite sure what you mean by taxon id . Can you please provide an example?

**Comment by myrmoteras** [comment]
can you please translate this into a format we use to work: please open the html version of the treatment and point us at the two wrong taxonomic names.

**Comment by jhpoelen** [comment]
@myrmoteras thanks for being patient with me as we are working our way through this tedious annotation adventure.

In reviewing https://tb.plazi.org/GgServer/html//C30587A9A562FF93FF2F350F875ED55F as accessed just now, it appears that the undesired occurrences of "Barbastello leucomelas" are **not** rendered in the html view onto the treatment. However, in the treatments-xml version, they do appear on lines 52 and 58. So, it appears I cannot point you to the text segment in the html version of the treatment, because the text is not translated into the html view.

Perhaps easier to point to a specific line in a github version at:

line 52 : 

https://github.com/plazi/treatments-xml/blob/d1dc6a5b4370612f90418409ced19be7a1352a36/data/C3/05/87/C30587A9A562FF93FF2F350F875ED55F.xml#L52

and line 58

https://github.com/plazi/treatments-xml/blob/d1dc6a5b4370612f90418409ced19be7a1352a36/data/C3/05/87/C30587A9A562FF93FF2F350F875ED55F.xml#L58

My question remains - 

What kind of specific reference will help us to point *exactly* to text segments that are subject of the annotation?

**Comment by jhpoelen** [comment]
I can confirm that this specific occurrence of Barbastello leucomelas was updated as suggested @kephelps. 

For more information see https://github.com/jhpoelen/msw-plazi/commit/905e30326b6123c09e867f8752dea3d065d63e37 .
